streamingpro-mlsql/src/main/java/streaming/dsl/mmlib/algs/SQLNaExt.scala

```diff
@@ -0,0 +1,74 @@
+package streaming.dsl.mmlib.algs
+
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import streaming.dsl.mmlib.SQLAlg
+import streaming.dsl.mmlib.algs.param.{BaseParams, WowParams}
+import streaming.session.MLSQLException
+
+/*
+ *
+ * Implement the spark DataFrameNaFunctions function
+ * https://spark.apache.org/docs/2.4.0/api/java/org/apache/spark/sql/DataFrameNaFunctions.html
+ *
+ */
+
+class SQLNaExt(override val uid: String) extends SQLAlg with WowParams {
+
+  override def train(df: DataFrame, path: String, params: Map[String, String]): DataFrame = {
+
+    val exe = params.get(execute.name).getOrElse(
+      "fill"
+    )
+
+    if (!execute.isValid(exe)) {
+      throw new MLSQLException(s"${execute.name} should be fill or drop or replace")
+    }
+
+    val inputTypeParam = params.get(inputType.name).getOrElse(
+      "string"
+    )
+    if (!execute.isValid(inputTypeParam)) {
+      throw new MLSQLException(s"${inputType.name} should be int or long or string or double")
+    }
+
+    if (exe == "fill") {
+      if (inputTypeParam == "long") {
+        val value = params.get("value").getOrElse(0L).asInstanceOf[Long]
+        df.na.fill(value)
+      } else if (inputTypeParam == "double") {
+        val value = params.get("value").getOrElse(0.0).asInstanceOf[Double]
+        df.na.fill(value)
+      } else if (inputTypeParam == "string") {
+        val value = params.get("value").getOrElse("").asInstanceOf[String]
+        df.na.fill(value)
+      }
+    } else if (exe == "drop") {
+      df.na.drop(params.get("value").getOrElse(""))
+    //TODO
+    //} else if(exe == "replace") {
+      //df.na.replace()
+    }
+
+    df
```

Comment:
`df.na .fill1 is in-place-operation?  maby `newdf = df.na.fill(value)` would be better?